Week3/homework/step2-6.js

```diff
@@ -4,18 +4,39 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArr = [];
+  arr.forEach(element => {
+    element.forEach(item => {
+      newArr.push(item);
+    });
+  });
+  return newArr;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  const newArr = [];
+  arr.forEach(element => {
+    element.forEach(item => {
+      item.forEach(elm => {
+        newArr.push(elm);
+      });
+    });
+  });
+  return newArr;
```

Comment:
Very well. Would you like to try it with `reduce` as well? I believe your code would be much smaller and easier to understand if using `reduce` rather than `forEach`. 🙂 

Also, if you like to take a step further, try to search on google on how you can implement it using recursion.

Reply:
Okay I added two functions using reduce and recursion.